src/Workspaces/CSharp/Portable/Formatting/Engine/Trivia/TriviaDataFactory.CodeShapeAnalyzer.cs

```diff
@@ -300,6 +297,17 @@ private bool ShouldFormat()
                 return false;
             }
 
+            private bool OnDisabledTextTrivia(SyntaxTrivia trivia, int index)
+            {
+                if (trivia.IsKind(SyntaxKind.DisabledTextTrivia) &&
+                    SyntaxFacts.IsNewLine(trivia.ToString().Last()))
```

Comment:
I don’t what it does, what I mean is, is it internally do string[string.length -1] or is it do

Foreach(var c in string)
{
… get to the last one.
Return ch;
}


-          Heejae

From: Balaji [mailto:notifications@github.com]
Sent: Thursday, February 19, 2015 3:49 PM
To: dotnet/roslyn
Cc: HeeJae Chang
Subject: Re: [roslyn] Format hash directives inside other hash directives of complex trivia (#654)


In src/Workspaces/CSharp/Portable/Formatting/Engine/Trivia/TriviaDataFactory.CodeShapeAnalyzer.cs<https://github.com/dotnet/roslyn/pull/654#discussion_r25038276>:

> @@ -300,6 +297,17 @@ private bool ShouldFormat()

>                  return false;

>              }

>

> +            private bool OnDisabledTextTrivia(SyntaxTrivia trivia, int index)

> +            {

> +                if (trivia.IsKind(SyntaxKind.DisabledTextTrivia) &&

> +                    SyntaxFacts.IsNewLine(trivia.ToString().Last()))

Last() is an extension method which gets the last char. I should introduce a NullEmpty check.

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/roslyn/pull/654/files#r25038276>.